typescript/lib/parsers/palm.ts

```diff
@@ -29,22 +69,75 @@ export class PaLMTextParser extends ParameterizedModelParser {
     options?: InferenceOptions | undefined,
     params?: JSONObject | undefined
   ): Promise<Output[]> {
-    throw new Error("Method not implemented.");
+    if (!this.client) {
+      const apiKey = getAPIKeyFromEnv("PALM_KEY");
+      this.client = new TextServiceClient({
+        authClient: new GoogleAuth().fromAPIKey(apiKey),
+      });
+    }
+
+    const completionParams = this.deserialize(prompt, aiConfig, params);
+    // Palm doesn't support streaming
+    const response = (await this.client.generateText(completionParams))[0];
+
+    const outputs: ExecuteResult[] = constructOutputs(response);
+
+    return outputs;
   }
 
-  public getOutputText(aiConfig: AIConfigRuntime, output?: Output | undefined, prompt?: Prompt | undefined): string {
-    throw new Error("Method not implemented.");
+  public getOutputText(aiConfig: AIConfigRuntime, output?: Output, prompt?: Prompt): string {
+    if (output == null && prompt != null) {
+      output = aiConfig.getLatestOutput(prompt);
+    }
+
+    if (output == null) {
+      return "";
+    }
+
+    if (output.output_type === "execute_result") {
+      return output.data as string;
+    } else {
+      return "";
+    }
```

Comment:
Let's add a TODO on GH issue to centralize all these getOutputText into a utils function. They're all so similar